pkg/ffapi/restfilter_json.go

```diff
@@ -210,7 +210,13 @@ func (jq *QueryJSON) addSimpleFilters(ctx context.Context, fb FilterBuilder, jso
 func joinShortNames(long, short, negated []*FilterJSONKeyValue) []*FilterJSONKeyValue {
 	res := make([]*FilterJSONKeyValue, len(long)+len(short)+len(negated))
 	copy(res, long)
-	copy(res[len(short):], short)
+
+	if len(long) == 0 {
+		copy(res[0:], short)
+	} else {
+		copy(res[len(short):], short)
```

Comment:
Shouldn't this be `len(long)` or `len(res)`? Since we want to copy into from index of where long left off to short... 
```suggestion
		copy(res[len(res):], short)
```

Reply:
This absolutely should be either `long` or `res` here (my bad) and the UT should have caught it but didn't, so will change that too.

This [comment](https://stackoverflow.com/a/58726780) indicates some potentially nasty behaviour with `append` and the existing specific structure here makes me worry for unintended consequences🔥

Reply:
Resolved in https://github.com/hyperledger/firefly-common/pull/124/commits/d549fecace2166b1738d25f90a9cd2ee97d6fe47, looking further at the code `short` as the index value is obviously wrong, but would actually never cause an issue since if you provide more than one `eq` value in a block, the other values are ignored. (This means the failure can't be recreated in a UT, since it's not actually a possible condition).